### PR TITLE
[YW] Added initContainer to copy the prometheus configuration

### LINKED ISSUE #7515: Add init container to copy prometheus config config map -> persistent volume
Instead of copying the default prometheus config to YW storage during YW startup, we should be using an init container to do so such that we can ensure the file has been copied before YW/prometheus containers startup.

### PR DESCRIPTION
**Pull request by baba230896**
### Summary

- Added `initContainer` in the YW helm chart to copy the Prometheus configuration.

### Test Plan

- Deployed the YW & spin up the YB universe on GCP. Updated the Prometheus configuration & is reflected in the Prometheus.

fixes: https://github.com/yugabyte/yugabyte-db/issues/7515

### CODE REVIEW

File: stable/yugaware/templates/service.yaml [hunk 1]
```diff
@@ -99,6 +99,22 @@ spec:
           secret:
             secretName: {{ .Release.Name }}-yugaware-tls-cert
         {{- end }}
+      initContainers:
+        - image: busybox:1.33.0
```
Comment (jvigil-yugabyte) [inline stable/yugaware/templates/service.yaml]: In the Yugabyte helm chart, we require busybox:1.31 (https://github.com/yugabyte/charts/blob/master/stable/yugabyte/templates/service.yaml#L392). Requiring two separate busybox images is a bit strange, especially for airgapped installations which will require the administrator to provide both. Should we use the same image in both places? Also, would we want to make this image + tag configurable via the values.yaml file like our other images? CC: @iSignal

Comment (jvigil-yugabyte) [reply stable/yugaware/templates/service.yaml]: Also, just noticed https://github.com/yugabyte/charts/pull/54, which removes the other busybox image and replaces it with the yugabyte image. Maybe we would we want to use the yugaware image here, then?

Comment (iSignal) [reply stable/yugaware/templates/service.yaml]: Agree that reusing an existing container would make it easier to deploy in offline environments where specific approved containers from an internal repo can be used.

**Review by iSignal — CHANGES_REQUESTED** [review]
use a diff base image

**Comment by baba230896** [comment]
@iSignal 
Could you please review it once whenever you've time? This is a very straightforward change.

File: stable/yugaware/templates/statefulset.yaml [hunk 2]
```diff
@@ -64,6 +64,22 @@ spec:
           secret:
             secretName: {{ .Release.Name }}-yugaware-tls-cert
         {{- end }}
+      initContainers:
```
Comment (skorobogatydmitry) [inline stable/yugaware/templates/statefulset.yaml:67]: Suggestion: https://github.com/elastic/helm-charts/blob/master/logstash/templates/statefulset.yaml#L149

This approach is more flexible.

File: stable/yugaware/templates/statefulset.yaml [hunk 3]
```diff
@@ -64,6 +64,22 @@ spec:
...
+        - image: {{ include "full_yugaware_image" . }}
+          name: prometheus-configuration
+          command: [ "/bin/sh", "-c"]
+          args:
+            - "cp /default_prometheus_config/prometheus.yml /prometheus_configs/prometheus.yml"
```
Comment (skorobogatydmitry) [inline stable/yugaware/templates/statefulset.yaml]: Why not use `cp` directly ?

File: stable/yugaware/templates/statefulset.yaml [hunk 4]
```diff
@@ -130,7 +146,7 @@ spec:
           {{- end }}
           command: [ "/bin/bash", "-c"]
           args:
-            - "cp /default_prometheus_config/prometheus.yml /prometheus_configs/prometheus.yml && bin/yugaware -Dconfig.file=/data/application.docker.conf"
```
Comment (skorobogatydmitry) [inline stable/yugaware/templates/statefulset.yaml:133]: So we don't need [this](https://github.com/yugabyte/charts/pull/63/files#diff-58c816a5badea1634d1f7fd6bc835152552ed86e143de953e41b0953a9313deeR197-R198) anymore, right ?

File: stable/yugaware/templates/statefulset.yaml (hunk 2 above)
Comment (iSignal) [reply stable/yugaware/templates/statefulset.yaml:67]: is the suggestion that we customize the init containers through values? For our current purpose, the init container here seems reasonable.

Comment (bhavin192) [reply stable/yugaware/templates/statefulset.yaml:67]: Yes, I agree with Sanketh. The current initContainer is something which is always going to be there in every chart installation.

File: stable/yugaware/templates/statefulset.yaml (hunk 3 above)
Comment (baba230896) [reply stable/yugaware/templates/statefulset.yaml]: Yes, we can use `cp` directly. Will update it.

File: stable/yugaware/templates/statefulset.yaml (hunk 4 above)
Comment (baba230896) [reply stable/yugaware/templates/statefulset.yaml:133]: I guess this is something I missed while rebasing to remove the merge conflicts. Indeed, I'll remove the `volumeMount` related to default Prometheus conf in YW STS Pod Spec.

**Comment by baba230896** [comment]
@skorobogatydmitry Could you please review it again?

File: stable/yugaware/templates/statefulset.yaml (hunk 2 above)
Comment (skorobogatydmitry) [reply stable/yugaware/templates/statefulset.yaml:67]: Yes, suggestion was to have `initContainers` in Values.
This allows to tune pods without re-building image and overall helpful.
And you're right - it's not required for this particular patch.

**Comment by skorobogatydmitry** [comment]
> @skorobogatydmitry Could you please review it again?

Sure!
There is "request review" button in `Reviewers` section - this works better to let reviewer know to re-visit PR.

**Review by skorobogatydmitry — APPROVED** [review]
LGTM! Please collect one more approval.

**Review by bhavin192 — APPROVED** [review]
This looks good to me!

**Pull request merged**